ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+from .skymodel import energyplus_horizontal_infrared
+
+from .psychrometrics import rel_humid_from_db_dpt
+from .psychrometrics import dew_point_from_db_hr
+from .psychrometrics import dew_point_from_db_enth
+from .psychrometrics import dew_point_from_db_wb
+
+import math
+import os
+import re
+import codecs
+import platform
+
+day_types = ['SummerDesignDay', 'WinterDesignDay', 'Sunday', 'Monday',
+             'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Holiday',
+             'CustomDay1', 'CustomDay2']
+h_types = ['Wetbulb', 'Dewpoint', 'HumidityRatio', 'Enthalpy']
+sky_types = ['ASHRAEClearSky', 'ASHRAETau', 'ZhangHuang', 'Schedule']
+comments = ['!- Name',
+            '!- Month',
+            '!- Day of Month',
+            '!- Day Type',
+            '!- Max Dry-Bulb Temp [C]',
+            '!- Daily Dry-Bulb Temp Range [C]',
+            '!- Dry-Bulb Temp Range Modifier Type',
+            '!- Dry-Bulb Temp Range Modifier Schedule Name',
+            '!- Humidity Condition Type',
+            '!- Wetbulb/Dewpoint at Max Dry-Bulb [C]',
+            '!- Humidity Indicating Day Schedule Name',
+            '!- Humidity Ratio at Maximum Dry-Bulb {kgWater/kgDryAir}',
+            '!- Enthalpy at Maximum Dry-Bulb {J/kg}',
+            '!- Daily Wet-Bulb Temperature Range [deltaC]',
+            '!- Barometric Pressure [Pa]',
+            '!- Wind Speed {m/s}',
+            '!- Wind Direction [Degrees; N=0, S=180]',
+            '!- Rain [Yes/No]',
+            '!- Snow on ground [Yes/No]',
+            '!- Daylight Savings Time Indicator',
+            '!- Solar Model Indicator',
+            '!- Beam Solar Day Schedule Name',
+            '!- Diffuse Solar Day Schedule Name',
+            '!- ASHRAE Clear Sky Optical Depth for Beam Irradiance (taub)',
+            '!- ASHRAE Clear Sky Optical Depth for Diffuse Irradiance (taud)',
+            '!- Clearness [0.0 to 1.2]']
+temp_multipliers = [0.82, 0.88, 0.92, 0.95, 0.98, 1, 0.98, 0.91, 0.74,
+                    0.55, 0.38, 0.23, 0.13, 0.05, 0, 0, 0.06, 0.14, 0.24,
+                    0.39, 0.5, 0.59, 0.68, 0.75]
+
+
+class Ddy(object):
+    """Import data from a local .ddy file.
+
+    properties:
+        location: A Ladybug location object
+        design_days: A list of the design days in the ddy file.
+        file_path: Optional file path into which ddy file can be written.
+    """
+    # TODO: set the default file path to use the ladybug default folder.
+    def __init__(self, location, design_days=[], file_path=None):
+        """Initalize the class."""
+        self.location = location
+        self.design_days = design_days
+        self.file_path = file_path or 'C:\\ladybug\\unnamed.ddy'
+
+    @classmethod
+    def from_ddy_file(cls, file_path):
+        """Initalize from a ddy file object from an existing ddy file.
+
+        args:
+            file_path: A full string to the .ddy file.
+        """
+        # defaults
+        location = None
```

Comment:
Why would you need that here?

Reply:
You are right.  This is not needed as long as I have something to catch the case that no location object is found in the .ddy file.